spec/term/1.0.md

```diff
@@ -0,0 +1,204 @@
+
+*Rationale:* To avoid divide-by-zero bugs in client code.
+
+### 3.2. The `term.viewport-height` property
+
+- Acceptable values: unsigned integers
+- Can be set by client: no
+
+The value of this property is the maximum number of lines that can be visible at once on the terminal viewport.
+A value of zero indicates that the viewport is infinitely tall.
+
+*Rationale:* Continuous stationery printers are a practical example of vertically unbounded viewports.
+
+### 3.3. The `term.scrollback-height` property
+
+- Acceptable values: unsigned integers
+- Can be set by client: no
+
+The value of this property is the average number of lines that can be retained in the terminal document.
+A value of zero indicates that the server cannot make a useful guess.
+
+This property is purely informational.
+A non-zero value does *not* mean that exactly that many lines will be retained in any particular situation.
+
+*Rationale:* When a lot of short lines are printed, typical implementations will be able to retain more scrollback then when a lot of long lines are printed.
+
+### 3.4. The `term.input-immediate` property
+
+- Acceptable values: booleans
+- Can be set by clients: yes
+
+This property is connection-lifetime-scoped, as defined above.
+The initial value is false.
+
+A value of true indicates that the server MUST make all input available to clients immediately, regardless of how many or which characters have been received.
+
+A value of false indicates that the server MAY buffer input and only make it available to clients once a certain condition has been met (most commonly, when an entire line of input has been received).
+
+### 3.5. The `term.input-echo` property
+
+- Acceptable values: booleans
+- Can be set by clients: yes
+
+This property is connection-lifetime-scoped, as defined above.
+The initial value is true.
+
+When the value of this property is true, all input that is made available to a client is also added to the terminal document as if its UTF-8-encoded form had been received by the server from a client's standard output.
+
+### 3.6. The `term.output-protected` property
+
+- Acceptable values: booleans
+- Can be set by clients: yes
+
+This property is connection-lifetime-scoped, as defined above.
+The initial value is false.
+
+The semantics of this property are defined in section 1.2.
+
+### 3.7. The `term.output-reflow` property
+
+- Acceptable values: booleans
+- Can be set by clients: no
+
+A value of true indicates that the server reflows the entire terminal document when the viewport width (i.e., the value of the `term.viewport-width` property) changes.
+
+A value of false indicates that the server MAY, after a viewport width change, continue to display unchanged parts of the terminal document according to the old viewport width.
+
+### 3.8. The `term.output-wordwrap` property
+
+- Acceptable values: booleans
+- Can be set by clients: no
```

Comment:
I don't see why we should forbid clients to set this.

Reply:
If this can be set by a client, we have to define how the layout works when fundamental layout properties can change over time. I do not find this a productive use of our time since I don't see a usecase for a client changing this property.

Reply:
Accepted.